### ISSUE TITLE
Multiples of ...

### ISSUE THREAD

**Issue by codonade** [codonade/mllabs#1]
Let's create a model that learns about the multiples of a certain number and, when it's confident enough, can give you the $`nth`$ multiple of that number - a truly exciting and game-changing project! Jokes aside, this is - in my opinion - a pretty good start in the vast world of machine learning, since it'll allow me to wrap my head around the very basics of algorithms and principles used in more serious implementations.

# 🕸️ Neural Network

Our whole neural network - if you can call it that - consists of a single neuron which receives a single input and produces, also, a single output - quite a simple formula: $`y = wx`$, where $`x`$ is the input to the neuron (the number of the multiple, $`n`$), $`w`$ is the weight the model has to learn, and $`y`$ is the output (the $`nth`$ multiple of $`w`$). In the machine learning language, this means that our neural network has a single parameter, $`w`$. In comparison, GPT-4 has around 1.76 trillion parameters...

# 🎯 Key Challenges

- **Teaching the model the weight $`(w)`$ efficiently:** Since this is a simple linear model, the main goal is for it to converge on the correct weight. This is the most crucial challenge of any model, and learning the core concepts of this will help us a lot when exploring more complicated ones.
- **Calculating the cost (sometimes referred to as the loss) of the model:** Basically telling the model how well or bad it's performing, so it can improve itself. This also involves choosing a cost function, but we'll probably go with the standard (Mean Squared Error) function.
- **Splitting the dataset into _training_ and _testing_ (or _validation_) data:** In a real machine learning setting, this ensures the model generalizes well to unseen inputs by splitting the dataset we have into training and testing sets (e.g., 80% for training, 20% for testing). This may not be entirely helpful to our simple model, but it's important nonetheless.
